pages/spacedeck/[[...roomId]].js

```diff
@@ -140,21 +142,51 @@ export default function Spacedeck() {
         };
     }, [spacedeck]);
 
-    async function createSketchRoom(link, name, parent = serviceSpaceId) {
-        // eslint-disable-next-line no-undef
-        logger.debug('creating room for ' + name);
-        const room = await matrix.createRoom(name, false, '', 'invite', 'content', 'spacedeck').catch(() => {
-            setErrorMessage(t('Something went wrong when trying to create a new room'));
-        });
-        await auth.getAuthenticationProvider('matrix').addSpaceChild(parent, room).catch(() => {
-            setErrorMessage(t('Couldn\'t add the new room to your sketch folder'));
-        });
-        await matrixClient.sendMessage(room, {
-            msgtype: 'm.text',
-            body: link,
-        }).catch(() => {
-            setErrorMessage(t('Something went wrong when trying to save the new sketch link'));
-        });
+    async function createSketchRoom(link, name, parent = serviceSpaceId, retries = 0) {
+        // Log debugging information
+        setUserFeedback(t('Syncing {{name}} from server', { name: name }));
+
+        // Create the room with retry handling
+        const createRoomForSketch = async (retries = 1) => {
+            logger.debug('Attempt %d of creating a room for %s', retries, name);
+
+            return await matrix.createRoom(name, false, '', 'invite', 'content', 'spacedeck');
+        };
+
+        const room = await createRoomForSketch()
+            .catch(async (error) => {
+                return matrix.handleRateLimit(error, () => createRoomForSketch())
+                    .catch(error => setErrorMessage(error.message));
+            });
+
+        // Log debug information about current progress
+        logger.debug('Created Room for %s with id %s ' + name, room);
```

Comment:
I think the `+` is wrong here. Also, when it comes to string interpolation I think I'd always prefer the ["new" version](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) of

```js
logger.debug(`Created room for ${name} with id ${room}`);
```